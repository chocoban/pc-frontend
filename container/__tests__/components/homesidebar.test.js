import { shallow, mount } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { Layout, Menu } from 'antd';

import HomeSidebar from '../../Home/Sidebar';

const { Sider } = Layout;

const initialState = {};
const mockStore = configureStore();
const store = mockStore(initialState);

describe('Sidebar tests', () => {
  let sidebarWrapper;
  beforeAll(() => {
    sidebarWrapper = mount(
      <Provider store={ store }>
        <HomeSidebar />
      </Provider>
    );
  });

  it('sidebar snapshot', () => {
    expect(sidebarWrapper).toMatchSnapshot();
  });

  it('collapses the sidebar', () => {
    sidebarWrapper
      .find('.side_bar')
      .at(0)
      .simulate('click');
    const siderbarDisplay = sidebarWrapper.find(Sider);
    expect(siderbarDisplay).toEqual({});
  });
});
