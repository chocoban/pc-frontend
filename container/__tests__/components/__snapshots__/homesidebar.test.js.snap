// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar tests sidebar snapshot 1`] = `ReactWrapper {}`;
